Title: "내가 작성 / 참여한 설문 목록 기능 개발 완료"

DevPCW opened the pull request:
**내용**
1) 내가 작성/참여한 설문 목록 조회
2) 내가 작성한 설문 삭제

Issue: #13

DevPCW committed:
[feat] :
1. 사용자 닉네임의 중복 여부를 확인 기능 추가
2. FileUploadUtil에서 FILE_PATH 경로 변경(react: public -> images) #7

DevPCW committed:
[feat] :
1. 사용자가 작성한 설문 목록을 조회 하는 기능
2. 사용자가 참여한 설문 목록을 조회 하는 기능
3. 사용자가 작성한 설문을 삭제하는 기능 #13

DevPCW committed:
[feat] :
불필요한 주석 및 코드 정리 #13

MyeoungDev commented on src/main/java/com/douzone/surveymanagement/mysurvey/controller/MySurveyApi.java:
```diff
@@ -0,0 +1,83 @@
+            @RequestBody MySurveyDTO mySurveyDTO) {
+
+        int deleted = mySurveyServiceImpl.updateMySurveysInProgress(mySurveyDTO);
+            if (deleted == 1) {
```
설문의 삭제가 됐는지 안됐는지를 1개가 됐는지 안됐는지를 통해서 확인하고 계시는데 이거는 service에서 처리를 하고 boolean을 return 받아서 처리하는게 더 깔끔할것 같습니다.

MyeoungDev commented on src/main/java/com/douzone/surveymanagement/mysurvey/dto/request/MySurveyDTO.java:
```diff
@@ -0,0 +1,36 @@
+ *
+ * @version 1.0
+ */
+@NoArgsConstructor
```
해당 dto에 기본생성자는 필요가 없는것 같아 보입니다!

MyeoungDev commented on src/main/java/com/douzone/surveymanagement/mysurvey/service/impl/MySurveyServiceImpl.java:
```diff
@@ -0,0 +1,42 @@
+        if (mySurveyDTO.getSurveyStatusNo() == 1) {
+            isDeleted = mySurveyMapper.updateMySurveysInProgress(mySurveyDTO);
+        }
+        return isDeleted;
```
해당 부분은 `updateSurveysInProgress` 메서드의 return 값을 통해 불필요한 변수를 선언하지 않고 한줄로 처리가 가능할꺼 같아보입니다.
시간이 되시면 수정해보시는것도 좋을것 같습니다!

MyeoungDev commented on src/main/java/com/douzone/surveymanagement/mysurvey/service/impl/MySurveyServiceImpl.java:
```diff
@@ -0,0 +1,42 @@
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
```
`Transactional(readonly=true)` 애너테이션을 선언함으로써 따로 `@Transactional`을 달아주지 않는 메서드는 `readonly` 로 동작하게 한다는 것을 명시적으로 선언해주는것도 좋아보입니다!

MyeoungDev commented on src/main/java/com/douzone/surveymanagement/mysurvey/service/impl/MySurveyServiceImpl.java:
```diff
@@ -0,0 +1,42 @@
+        List<MySurveyDTO> surveys = mySurveyMapper.selectMySurveysWithSorting(userNo);
+
+        return mySurveyMapper.selectMySurveysWithSorting(userNo);
+    }
```
현재 코드를 보면 동일한 동작을 하는것이 두번 실행이 되고,
맨 위의 `surveys` 는 아무런 동작도 하지 않는것으로 보입니다.
해당 부분은 수정 부탁드리겠습니다.

MyeoungDev commented on src/main/java/com/douzone/surveymanagement/user/controller/UserApi.java:
```diff
@@ -89,6 +96,23 @@ public ResponseEntity<UserDTO> getUserByUserNo(@PathVariable("userNo") long user
+        boolean isDuplicate = userServiceImpl.duplicateUsername(userModifyDTO);
+
+        if(isDuplicate) {
+            return ResponseEntity.status(HttpStatus.CONFLICT).body("Nickname is not available");
```
에러 상황에 맞는 적절한 status 를 보내주는 것은 좋은 방법으로 보입니다!

MyeoungDev commented on src/main/java/com/douzone/surveymanagement/user/dto/request/UserModifyDTO.java:
다른 부분은 validation이 전혀 필요하지 않아서 다른 validation은 진행하지 않은건가요?

MyeoungDev commented on src/main/java/com/douzone/surveymanagement/user/service/impl/UserServiceImpl.java:
```diff
@@ -28,6 +31,12 @@ public class UserServiceImpl implements UserService {
 
     @Override
```
`update` 메서드로 보이는데 트랜잭션이 걸려있지 않은것 같습니다!

MyeoungDev commented on src/main/java/com/douzone/surveymanagement/user/service/impl/UserServiceImpl.java:
```diff
@@ -37,6 +46,24 @@ public UserDTO getUserByUserNo(long userNo) {
+        UserModifyDTO existUserNickname = myPageMapper.getUserByUserNickname(userModifyDTO.getUserNickname());
+
+        if (existUserNickname != null && isUserNicknameDuplicate(existUserNickname.getUserNickname())) {
+            return true;
```
해당 `if` 문안에서의 동작이 모두 동일한 동작을 하는것으로 보입니다.
다른 의도가 있는건가요?

MyeoungDev commented on src/main/resources/mappers/mySurveyMapper.xml:
현재 모든 쿼리에  <!CDATA[]> 를 선언해 두었지만, 해당 선언이  불필요해 보입니다.
<!CDATA[]>  대해서 알아보는것은 어떨까요?

MyeoungDev replied:
파일의 이름은 대문자로 시작하는것은 어떨까요?

MyeoungDev commented on src/main/resources/mappers/userMapper.xml:
`MySurveyMapper` 파일에는 들여쓰기와 대소문자 구분이 잘 되어있어 읽기 편했습니다.
해당 코드들에도 위와 같이 작성하면 더 읽기 좋은 코드가 될 것 같습니다.

MyeoungDev reviewed (commented):
고생하셨습니다 👍

ajinss reviewed (approved):
수고하셨습니다!!

DevPCW committed:
[feat] :
1. 작성 중 설문 삭제 기능에서 boolean을 return
2. Transactional(readonly=true) 애너테이션을 선언
3. updateSurveysInProgress 메서드의 return 값을 통해 불필요한 변수선언 수정
4. duplicateUsername 메소드 if문 수정
5. DTO 기본 생성자 애너테이션 삭제
6. xml 클린코드
7. 그외 코드 리팩토링 #13

kl204 reviewed (approved):
고생하셨습니다!
